contrib/gcppubsub/pkg/dispatcher/receiver/receiver.go

```diff
@@ -152,3 +172,36 @@ func (r *Receiver) getChannel(ctx context.Context, ref provisioners.ChannelRefer
 
 	return c, err
 }
+func (r *Receiver) getHostToChannelMap() map[string]provisioners.ChannelReference {
+	return r.hostToChannelMap.Load().(map[string]provisioners.ChannelReference)
+}
+
+func (r *Receiver) setHostToChannelMap(hcMap map[string]provisioners.ChannelReference) {
+	r.hostToChannelMap.Store(hcMap)
+}
+
+// UpdateHostToChannelMap will be called from the controller that watches gcp-pubsub channels.
+// It will update internal hostToChannelMap which is used to resolve the hostHeader of the
+// incoming request to the correct ChannelReference in the receiver function.
+func (r *Receiver) UpdateHostToChannelMap(ctx context.Context) error {
+	logging.FromContext(ctx).Info("UpdateHostToChannelMap: Acquiring mutex lock")
```

Comment:
This function seems to have excessive logging. I think you drop this one and "acquired mutex lock". If you want to keep them, make them Debug.